spring-boot-admin-server/src/main/java/de/codecentric/boot/admin/notify/MicrosoftTeamsNotifier.java

```diff
@@ -0,0 +1,372 @@
+package de.codecentric.boot.admin.notify;
+
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import de.codecentric.boot.admin.event.ClientApplicationDeregisteredEvent;
+import de.codecentric.boot.admin.event.ClientApplicationEvent;
+import de.codecentric.boot.admin.event.ClientApplicationRegisteredEvent;
+import de.codecentric.boot.admin.event.ClientApplicationStatusChangedEvent;
+import de.codecentric.boot.admin.model.Application;
+import de.codecentric.boot.admin.model.StatusInfo;
+import org.springframework.web.client.RestTemplate;
+
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.MissingFormatArgumentException;
+
+public class MicrosoftTeamsNotifier extends AbstractEventNotifier {
+
+    public static final String STATUS_KEY = "Status";
+    public static final String SERVICE_URL_KEY = "Service URL";
+    public static final String HEALTH_URL_KEY = "Health URL";
+    public static final String MANAGEMENT_URL_KEY = "Management URL";
+    public static final String SOURCE_KEY = "Source";
+    private final RestTemplate restTemplate;
+
+    /**
+     * Webhook url for Microsoft Teams Channel Webhook connector
+     *  (i.e. https://outlook.office.com/webhook/{webhook-id})
+     */
+    private URI webhookUrl;
+    /**
+     * Image url for image displayed in activity section
+     *  (i.e. https://raw.githubusercontent.com/tomd8451/spring-boot-admin
+     *          /master/spring-boot-admin-server-ui/core/img/platform-spring-boot.jpg)
+     */
+    private URI imageUrl = URI.create("https://raw.githubusercontent.com/tomd8451/spring-boot-admin" +
```

Comment:
I still don't like the fact that this points to your private repo. Would it be possible to use a smaller png (or even better a svg) and encode it as data url?

Reply:
Microsoft Teams isn't properly rendering a data url. It works when viewed in a browser, but not on the mobile app/client application. An icon is configurable for the webhook when it's created, so just removing the icon from the message.  Addressed this in [0f7e1643d589b5e93918d5c7c167516c308cc868](https://github.com/codecentric/spring-boot-admin/pull/592/commits/0f7e1643d589b5e93918d5c7c167516c308cc868)